src/library_int53.js

```diff
@@ -111,5 +110,34 @@ mergeInto(LibraryManager.library, {
   // TODO: Add $convertU32PairToI53Signaling() variant.
   $convertU32PairToI53: function(lo, hi) {
     return (lo >>> 0) + (hi >>> 0) * 4294967296;
-  }
+  },
+
+  // Same as convertI32PairToI53 but returns NaN if the pair of I32 numbers
+  // combine to form a number outside the i53 range
+  // (MIN_SAFE_INTEGER - MAX_SAFE_INTEGER)
+  $convertI32PairToI53Signaling__deps: ['$convertI32PairToI53'],
+  $convertI32PairToI53Signaling: function(lo, hi) {
```

Comment:
I think this would probably be better named "clamping" or something rather than "signaling". I think of "signaling" as more like throwing (e.g. like a signaling NaN).

Actually I guess "clamping" wouldn't be right either since that would imply returning the max value rather than a sentinel value. Not sure what a better idea would be.

Reply:
Don't you think that returns NaN counts as a "signal"?   Maybe "Checked" would be a better adjective, but the other ones in this files end in `Signaling`.. but then they also do actually throw.

Reply:
Maybe `convertI32PairToI53OrNaN`? The "I53" hints that it would return an integer, but then the function can also return a non-integer (NaN)

Reply:
I prefer `convertI32PairToI53Checked` myself since it tells me what the function is doing..  but since its a mostly-internal-facing function is don't feel strongly.. I will rename if you do.

Reply:
> since its a mostly-internal-facing function

The API in library_int53.js is inteded to be a public API for end users. It is common that end users also need to do i64/i53/i32 pair/Number marshalling across the boundary, so it is good that the API that the system libraries use would be the same that is exposed to end users.

Authored PRs #16865 and #16866 to implement a lean `$convertI32PairToI53Checked()` that is tested using the same machinery that the rest of library_int53.js is. Maybe that can be helpful?

Reply:
> I prefer convertI32PairToI53Checked myself since it tells me what the function is doing

I find it is odd to have the name "ToI53" there, when the result might not be an integer, and the name "checked" seems a bit ambiguous as to what kind of check is performed, and what the result of that check will be. Though maybe the "OrNaN" name would not either tell when exactly the NaN would appear, so guess that name is not much better either. Proposed the `Checked` name in the above PR.